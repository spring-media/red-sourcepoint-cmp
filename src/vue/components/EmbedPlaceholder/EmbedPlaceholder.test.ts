import Vue from 'vue';
import { mount } from '@vue/test-utils';
import { EmbedPlaceholder } from './';

const loadPrivacyManagerModal = jest.fn();

const privacyManagerStub = Vue.extend({
  name: 'PrivacyManager',
  render() {
    return (
      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
      this.$scopedSlots.default!({
        loadPrivacyManagerModal,
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
      }) as any
    );
  },
});

const customConsent = jest.fn();

const consentActionsStub = Vue.extend({
  name: 'ConsentActions',
  render() {
    return (
      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
      this.$scopedSlots.default!({
        customConsent,
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
      }) as any
    );
  },
});

describe('EmbedPlaceholder', () => {
  afterEach(() => {
    loadPrivacyManagerModal.mockReset();
  });

  it('should render without any errors', () => {
    expect(
      mount(EmbedPlaceholder, { propsData: { customConsents: {}, privacyManagerId: 123 } }).element,
    ).toMatchSnapshot();
  });

  it('should give a consent to given custom consents', () => {
    const customConsents = {};

    const wrapper = mount(EmbedPlaceholder, {
      stubs: { ConsentActions: consentActionsStub },
      propsData: { customConsents, privacyManagerId: 123 },
    });

    wrapper.find('.embed-placeholder__button').trigger('click');

    expect(customConsent).toHaveBeenCalledWith(customConsents);
  });

  it('should open a privacy manager by clicking on the link in the footer', () => {
    const wrapper = mount(EmbedPlaceholder, {
      stubs: { PrivacyManager: privacyManagerStub },
      propsData: { privacyManagerId: 12345, customConsents: {} },
    });

    wrapper.find('.embed-placeholder__text-link').trigger('click');

    expect(loadPrivacyManagerModal).toHaveBeenCalledWith(12345);
    expect(loadPrivacyManagerModal).toHaveBeenCalledTimes(1);
  });
});
