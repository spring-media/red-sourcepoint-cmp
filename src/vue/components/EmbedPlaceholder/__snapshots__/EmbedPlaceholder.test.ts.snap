// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedPlaceholder should render without any errors 1`] = `
<div
  class="embed-placeholder__container"
  data-vendor-id="abc"
>
  <div
    class="embed-placeholder__header"
  >
    <svg
      fill="none"
      height="80"
      viewBox="0 0 80 80"
      width="80"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M46.1772 43.5439L64.8503 62.217C67.8858 58.824 70.2367 54.8056 71.6858 50.3789L46.1772 43.5439Z"
        fill="#CED4DA"
        opacity="0.4"
      />
       
      <path
        d="M72.8214 45.8522L42.6667 37.7723V6.77197C59.8294 8.13047 73.3334 22.4883 73.3334 40.0002C73.3334 41.9967 72.8214 45.8522 72.8214 45.8522Z"
        fill="#CED4DA"
        opacity="0.6"
      />
       
      <path
        d="M37.3334 6.77197C20.1707 8.13047 6.66675 22.4883 6.66675 40.0002C6.66675 58.4097 21.5906 73.3335 40.0001 73.3335C48.1951 73.3335 55.6994 70.3762 61.504 65.4706L37.3669 41.3335H37.3334V6.77197Z"
        fill="#CED4DA"
      />
    </svg>
  </div>
   
  <div
    class="embed-placeholder__headline"
  >
     An dieser Stelle findest du Inhalte von Drittanbietern 
  </div>
   
  <div
    class="embed-placeholder__description"
  >
    
      Um mit Inhalten von Drittanbietern zu interagieren oder diese darzustellen, brauchen wir deine Zustimmung.
    
  </div>
   
  <button
    class="embed-placeholder__button"
  >
     externen Inhalt aktivieren 
  </button>
   
  <div
    class="embed-placeholder__footer-text"
  >
    
        Ich bin damit einverstanden, dass mir Inhalte von Drittanbietern angezeigt werden. Damit können
        personenbezogene Daten an Drittanbieter übermittelt werden. Dazu ist ggf. die Speicherung von Cookies auf
        deinem Gerät notwendig. Mehr Informationen dazu findest du
        
    <a
      class="embed-placeholder__text-link"
      href="#"
      rel="noopener noreferrer"
      target="_blank"
    >
      hier
    </a>
    .
      
  </div>
</div>
`;
